### PR TITLE
db_generator command line arguments

### LINKED ISSUE #15940: db_generator: substandard usability.
```
milovidov@milovidov-desktop:~/work/ClickHouse$ build/utils/db-generator/query_db_generator 
^C
milovidov@milovidov-desktop:~/work/ClickHouse$ build/utils/db-generator/query_db_generator --help
^C
milovidov@milovidov-desktop:~/work/ClickHouse$ build/utils/db-generator/query_db_generator < query.sql 
milovidov@milovidov-desktop:~/work/ClickHouse$ build/utils/db-generator/query_db_generator query.sql 
^C
```

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added boost::program_options to `db_generator` in order to increase its usability. This closes #15940

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Will wait only for builds.

File: utils/db-generator/query_db_generator.cpp [hunk 1]
```diff
@@ -1264,8 +1269,41 @@ TableList getTablesFromSelect(std::vector<std::string> queries)
...
+        if (vm.count("help") || vm.count("h"))
+        {
+            std::cout << "Hello! It is datasets generator for ClickHouse's queries." << std::endl;
+            std::cout << "Put some query as an input and it will produce queries for table creating and filling." << std::endl;
+            std::cout << "After that your query could be executed on this tables." << std::endl;
+            std::cout << desc << std::endl;
+            return 1;
+        }
+        if (vm.count("input"))
+            freopen(vm["input"].as<std::string>().c_str(), "r", stdin);
```
Comment (alexey-milovidov) [inline utils/db-generator/query_db_generator.cpp:1295]: BTW, I tried with stdin but it didn't work either.

**Review by alexey-milovidov — APPROVED** [review]

File: utils/db-generator/query_db_generator.cpp (hunk 1 above)
Comment (nikitamikhaylov) [reply utils/db-generator/query_db_generator.cpp:1295]: Maybe there were no semicolon at the end of query. Or you didn't press Ctrl-d..

```
☁️ ☁️ ☁️ @jakalletti-build:~/ClickHouse/build/memory/utils/db-generator  
↳ $ cat input.sql 
SELECT a from b;
☁️ ☁️ ☁️ @jakalletti-build:~/ClickHouse/build/memory/utils/db-generator  
↳ $ ./query_db_generator -i input.sql
CREATE DATABASE IF NOT EXISTS ;

CREATE TABLE IF NOT EXISTS b (
a Int64
) ENGINE = Log;

INSERT INTO b
(a) VALUES 
(495582894);

SELECT a from b; 
☁️ ☁️ ☁️ @jakalletti-build:~/ClickHouse/build/memory/utils/db-generator  
↳ $ ./query_db_generator 
Copy your queries (with semicolons) here, press Enter and Ctrl+D.
SELECT a from b;
CREATE DATABASE IF NOT EXISTS ;

CREATE TABLE IF NOT EXISTS b (
a Int64
) ENGINE = Log;

INSERT INTO b
(a) VALUES 
(495582894);

SELECT a from b; 

```

**Comment by nikitamikhaylov** [comment]
`heap-use-after-free` in Stress test is known.

File: utils/db-generator/query_db_generator.cpp [hunk 2]
```diff
@@ -1264,8 +1269,41 @@ TableList getTablesFromSelect(std::vector<std::string> queries)
...
+        if (vm.count("output"))
+            freopen(vm["output"].as<std::string>().c_str(), "w", stdout);
```
Comment (TCeason) [inline utils/db-generator/query_db_generator.cpp:1297]:
```
FAILED: utils/db-generator/CMakeFiles/query_db_generator.dir/query_db_generator.cpp.o 
/usr/bin/ccache /usr/bin/g++-10  -DAWS_SDK_VERSION_MAJOR=1 -DAWS_SDK_VERSION_MINOR=7 -DAWS_SDK_VERSION_PATCH=231 -DENABLE_OPENSSL_ENCRYPTION -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_EPOLL -DPOCO_OS_FAMILY_UNIX -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -DUNALIGNED_OK -DUSE_FASTMEMCPY=1 -DUSE_REPLXX=1 -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -I../contrib/sentry-native/include -Iincludes/configs -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -isystem ../contrib/libunwind/include -isystem ../contrib/msgpack-c/include -isystem ../contrib/aws/aws-cpp-sdk-s3/include -isystem ../contrib/aws/aws-cpp-sdk-core/include -isystem ../contrib/libhdfs3/include -isystem ../contrib/re2 -I../src -Isrc -Isrc/Core/include -I../base/common/.. -Ibase/common/.. -I../contrib/cityhash102/include -isystem ../contrib/boost -I../contrib/FastMemcpy -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Foundation/include -isystem ../contrib/poco/NetSSL_OpenSSL/include -isystem ../contrib/poco/Crypto/include -isystem ../contrib/openssl-cmake/linux_x86_64/include -isystem ../contrib/openssl/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/JSON/include -isystem ../contrib/poco/XML/include -isystem ../contrib/replxx/include -I../contrib/cctz/include -isystem ../contrib/fmtlib-cmake/../fmtlib/include -isystem ../contrib/double-conversion -isystem ../contrib/ryu -isystem contrib/re2_st -Icontrib/zlib-ng -I../contrib/zlib-ng -I../base/pcg-random/. -isystem ../contrib/croaring/include -I../contrib/aws-c-common/include -I../contrib/aws-c-event-stream/include -Icontrib/aws-s3-cmake/include -isystem ../contrib/libc-headers/x86_64-linux-gnu -isystem ../contrib/libc-headers -fdiagnostics-color=always -std=c++2a -fsized-deallocation  -pipe -msse4.1 -msse4.2 -mpopcnt   -Wall  -Werror -Wextra -Wframe-larger-than=32768 -O2 -g -DNDEBUG -O3  -fno-pie   -D OS_LINUX -Wbool-compare -Wcast-align -Wcast-qual -Wdelete-incomplete -Wdisabled-optimization -Wduplicated-cond -Wenum-compare -Winit-self -Wlogical-not-parentheses -Wlogical-op -Wmaybe-uninitialized -Wmisleading-indentation -Wmissing-include-dirs -Wnon-virtual-dtor -Wno-return-local-addr -Wodr -Wold-style-cast -Wplacement-new=2 -Wpointer-arith -Wredundant-decls -Wreorder -Wshadow -Wshift-negative-value -Wsized-deallocation -Wsizeof-array-argument -Wsizeof-pointer-memaccess -Wsuggest-override -Wswitch-bool -Wtautological-compare -Wtrampolines -Wunused -Wvector-operation-performance -Wzero-as-null-pointer-constant -Wno-sequence-point -Wno-stringop-overflow -nostdinc++ -MD -MT utils/db-generator/CMakeFiles/query_db_generator.dir/query_db_generator.cpp.o -MF utils/db-generator/CMakeFiles/query_db_generator.dir/query_db_generator.cpp.o.d -o utils/db-generator/CMakeFiles/query_db_generator.dir/query_db_generator.cpp.o -c ../utils/db-generator/query_db_generator.cpp
../utils/db-generator/query_db_generator.cpp: In function 'int main(int, const char**)':
../utils/db-generator/query_db_generator.cpp:1295:20: error: ignoring return value of 'FILE* freopen(const char*, const char*, FILE*)' declared with attribute 'warn_unused_result' [-Werror=unused-result]
 1295 |             freopen(vm["input"].as<std::string>().c_str(), "r", stdin);
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../utils/db-generator/query_db_generator.cpp:1297:20: error: ignoring return value of 'FILE* freopen(const char*, const char*, FILE*)' declared with attribute 'warn_unused_result' [-Werror=unused-result]
 1297 |             freopen(vm["output"].as<std::string>().c_str(), "w", stdout);
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors

```

When I try to build this pr, It returns the error.

My instances config:

```
➜  build git:(master) ✗ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

➜  build git:(master) ✗ cmake --version 
cmake version 3.10.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
➜  build git:(master) ✗ ninja --version 
1.8.2

```

And I use gcc-10 and g++-10 . So do you have any suggestions? @nikitamikhaylov @alexey-milovidov

Comment (TCeason) [reply utils/db-generator/query_db_generator.cpp:1297]: Oh, I have resolved it.

Comment (RaghavSood) [reply utils/db-generator/query_db_generator.cpp:1297]: @TCeason Was the resolution simply not treating warnings as errors, or something else?

Comment (TCeason) [reply utils/db-generator/query_db_generator.cpp:1297]: https://github.com/ClickHouse/ClickHouse/pull/16859

Have a look.@RaghavSood

Comment (RaghavSood) [reply utils/db-generator/query_db_generator.cpp:1297]: Splendid, thank you!

**Pull request merged**